purchase_request_to_procurement/models/purchase_request.py

```diff
@@ -73,6 +73,7 @@ def _prepare_procurement_order_values(
             'product_qty': self.product_qty,
             'product_uom': self.product_uom_id.id,
             'date_planned': self.date_required,
+            'request_id': self.request_id.id,
```

Comment:
Yes you are right. The link already exists, I missed that.

For the second point, I don't like to set a value on a field which might not exist ^^, in my opinion this is a programming error (even if there is only a warning). But as I said, this is just my opinion. And ```purchase_request_to_procurement``` should not inherit from ```purchase_request_procurement``` so we can't add this dependency.